src/components/CartProvider/utils.ts

```diff
@@ -37,13 +37,21 @@ export const extractCartLines = (
       return {
         ...node,
         quantity,
-        totalPrice: priceToString(
-          {
+        totalPrice: {
+          ...node.pricing.price,
+          currency: node.pricing.price.gross.currency,
+          gross: {
             amount: quantity * node.pricing.price.gross.amount,
-            currency: node.pricing.price.gross.currency,
+            ...node.pricing.price.gross,
+            localized: null,
           },
-          locale
-        ),
+          locale,
+          net: {
+            amount: quantity * node.pricing.price.net.amount,
+            ...node.pricing.price.net,
+            localized: null,
```

Comment:
It doesn't make sense to pass legacy localized field as null - probably my fault introduced in my commits...